README.md

```diff
@@ -1,77 +1,337 @@
+```
+
+We'll get to what the `module.exports` stuff is in just a second. We can import this component in our `index.js` by using
+`require()` and assigning the result to a variable:
+
+```js
+const component = require('./components/foo');
 ```
 
-(Or wherever you'd prefer to teach them to put the compiled JavaScript — be sure
-to add this file to the `.gitignore`.)
+Note that files are always referred to using a relative path (even if they are in the same directory). This way Node
+knows whether to look for a local module or one found in `node_modules`, or in the global modules. Adding the `.js` 
+extension is not required.
+
+Back to the exporting stuff! Using CommonJS, we have two options of exporting things out of our files: either through
+named exports (exporting multiple things) or a default export (exporting one thing).
+
+
+### Named exports
+Named exports allow us to export several things at once. This is useful for utility modules or libraries. Exporting
+several things at once is done by exporting an object when setting the value of `module.exports`.
+
+```js
+// In a file called `fruits.js`
+module.exports = {
+  apple: 'red',
+  banana: 'yellow',
+};
+
+// In a file in the same directory
+const fruits = require('./fruits');
+console.log(fruits.apple); // prints 'red'
 
-This way, what they're doing in `index.js` might look like
+// In another file, also in the same directory
+const apple = require('./fruits').apple;
+console.log(apple); // prints 'red'
+```
 
-```javascript
-const React = require('react')
-const ReactDOM = require('react-dom')
-const HelloWorld = require('./components/HelloWorld')
+When using named exports, we can choose to either import the entire thing and then reference the keys on the exported
+object, or we can import one specific key. The last example can also be rewritten using ES2015 destructuring syntax:
 
-ReactDOM.render(<HelloWorld />, document.getElementById('main'))
+```js
+const { apple } = require('./fruits');
+console.log(apple); // prints 'red'
 ```
 
-as long as `components/HelloWorld.js` looks like
+### Default export
+A default export means we're exporting just one thing. This is useful for exporting components in their own file, since
+there's only one thing there: the component itself. Exporting one thing only is done by exporting a reference to what
+we want to export when setting the value of `module.exports`. You can also inline the value of what you want to export.
 
-```javascript
-const React = require('react')
+```js
+// In a file called `Tweet.js`
+const React = require('react');
 
-class HelloWorld extends React.Component {
+class Tweet extends React.Component {
   render() {
-    return <h1>Hello, world!</h1>
+    return (
+      <div className="tweet">
+        <img src="http://twitter.com/some-avatar.png" className="tweet__avatar" />
+        <div className="tweet__body">
+            <p>We're writing this tweet in JSX. Holy moly!</p>  
+        </div>
+      </div>
+    );
```

Comment:
Seeing that mysterious highlighting again:

![](https://s3.amazonaws.com/ezmiller/public/images/learn-co-curriculumreact-jsx_at_wip-master_2016-09-09_17-19-07.png)